src/test/java/com/linagora/james/mailets/MoveMailetTest.java

```diff
@@ -0,0 +1,110 @@
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    private MailboxManager mailboxManager = null;
+    private MailboxId.Factory mailboxIdFactory = null;
+    private UsersRepository usersRepository = null;
+
+    @Test
+    public void initShouldThrowWhenHeaderNameIsEmpty() throws Exception {
+        expectedException.expect(MailetException.class);
+        expectedException.expectMessage("'headerName' is mandatory");
+
+        FakeMailetConfig config = FakeMailetConfig.builder()
+                .setProperty(MoveMailet.HEADER_NAME, "")
+                .build();
+        
+        MoveMailet testee = new MoveMailet(mailboxManager, mailboxIdFactory, usersRepository);
+        testee.init(config);
+    }
+
+    @Test
+    public void initShouldThrowWhenThresholdIsNull() throws Exception {
+        expectedException.expect(MailetException.class);
+        expectedException.expectMessage("'threshold' is mandatory");
+
+        FakeMailetConfig config = FakeMailetConfig.builder()
+                .build();
+        
+        MoveMailet testee = new MoveMailet(mailboxManager, mailboxIdFactory, usersRepository);
+        testee.init(config);
+    }
+
+    @Test
+    public void initShouldThrowWhenThresholdIsEmpty() throws Exception {
+        expectedException.expect(MailetException.class);
+        expectedException.expectMessage("'threshold' is mandatory");
+
+        FakeMailetConfig config = FakeMailetConfig.builder()
+                .setProperty(MoveMailet.THRESHOLD, "")
+                .build();
+        
+        MoveMailet testee = new MoveMailet(mailboxManager, mailboxIdFactory, usersRepository);
+        testee.init(config);
+    }
+
+    @Test
+    public void headerNameShouldEqualsPropertyWhenGiven() throws Exception {
+        FakeMailetConfig config = FakeMailetConfig.builder()
+                .setProperty(MoveMailet.HEADER_NAME, "my header")
+                .setProperty(MoveMailet.THRESHOLD, "98.3")
+                .build();
+        
+        MoveMailet testee = new MoveMailet(mailboxManager, mailboxIdFactory, usersRepository);
+        testee.init(config);
+        
+        assertThat(testee.headerName).isEqualTo("my header");
+    }
+
+    @Test
+    public void thresholdShouldEqualsPropertyWhenGiven() throws Exception {
+        FakeMailetConfig config = FakeMailetConfig.builder()
+                .setProperty(MoveMailet.HEADER_NAME, "my header")
+                .setProperty(MoveMailet.THRESHOLD, "98.3")
+                .build();
+        
+        MoveMailet testee = new MoveMailet(mailboxManager, mailboxIdFactory, usersRepository);
+        testee.init(config);
+        
+        assertThat(testee.threshold).isEqualTo(98.3);
+    }
+
+    @Test
+    public void getMailetInfoShouldReturnMailetName() {
+        MoveMailet testee = new MoveMailet(mailboxManager, mailboxIdFactory, usersRepository);
+        assertThat(testee.getMailetInfo()).isEqualTo("MoveMailet Mailet");
+    }
```

Comment:
No unit test for service?

What happe when we have a header? When we don't have it? When it's invalid? When various type of error are returned?

Can threashold be invalid? Negative? Null?